### ISSUE TITLE
Fill in test gaps for Event Pipe

### ISSUE THREAD

**Issue by tommcdon** [dotnet/coreclr#25090]
We need to add new tests to cover .net core 3.0 event pipe test scenarios.  These new tests will supercede the following:

- [ ] https://github.com/dotnet/coreclr/issues/23224
- [ ] https://github.com/dotnet/coreclr/issues/22891
- [ ] https://github.com/dotnet/coreclr/issues/20510
- [ ] https://github.com/dotnet/coreclr/issues/20299

**Comment by josalem** [comment]
Do we have a list of scenarios we want to prioritize?  I have a template for writing the tests that we can use to _hopefully_ iterate quickly through them.

**Comment by tommcdon** [comment]
Let's use this issue to track the sceanrios we wish to fill
@cshung @sywhang

**Comment by josalem** [comment]
I believe we should add tests covering the following "knobs":
* [ ] Messages sent vs received under:
  * varying buffer sizes
  * varying providers
* [ ] Provider turned on vs sane traces (e.g., if I trace GC, I should see equal numbers of start and end events)
  * varying buffer sizes
* [ ] EventPipe Connection _and_ EventListener connection being coherent

**Comment by jorive** [comment]
@josalem Should we add to the mix:

- [ ] Verify providers/events enabled are written
- [ ] Verify CPU stacks are written when enabled
- [ ] Handle erroneous user input

**Comment by josalem** [comment]
Agreed.  I think we should have a small base set of tests we can add to PR validation, and a larger, more complex set we can add to pri-1 testing.  These tests give us a chance to implicitly "soft" document some guarantees about the EventPipe system that shouldn't be broken by incoming changes.

**Comment by noahfalk** [comment]
What do you guys think about promoting this list of test cases into a markdown doc that we put in the tree? Often when researching tests available in some area I'd run into two questions:
a) Is scenario X tested?
b) If it is, which of the 100 tests in this area covers it?
Typically this is looking back months/years after the tests were originally written so any issues (such as this one) aren't easily discoverable at that point.

**Comment by sywhang** [comment]
Sounds good to me.
